perf(rolldown): take advantage of tokio blocking threads

Brooooooklyn opened the pull request:
closes https://github.com/rolldown/rolldown/issues/3754

Benchmark in apps/10000:

### `1.0.0-beta.38`

```
❯ hyperfine --warmup 3 --runs 10 --command-name "rolldown-latest" "../../node_modules/.bin/rolldown build --config rolldown.config.mjs"
Benchmark 1: rolldown-latest
  Time (mean ± σ):     553.6 ms ±  21.4 ms    [User: 1107.1 ms, System: 2948.2 ms]
  Range (min … max):   518.7 ms … 579.5 ms    10 runs
```

### `current`

```
❯ hyperfine --warmup 3 --runs 10 --command-name "rolldown-latest" "../../node_modules/.bin/rolldown build --config rolldown.config.mjs"
Benchmark 1: rolldown-latest
  Time (mean ± σ):     458.1 ms ±   6.5 ms    [User: 1052.2 ms, System: 1092.4 ms]
  Range (min … max):   447.5 ms … 468.8 ms    10 runs
```

I don't know why `codspeed` crashes, I added a feature for `codspeed` so that under `codspeed` it doesn't use `spawn_blocking`.

Brooooooklyn commented:
* **#6306** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6306?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#6305** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6305?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#6272** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6272?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#6271** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6271?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#6270** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6270?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6270?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>graphite: merge</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/rolldown?ref=merge-queue-instructions-in-stack-comment&prId=6270).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

CPunisher commented:
codspeed crashes because it only support up to 500 threads. So you also need to limit the max block thread count in benchmark code.

IWANABETHATGUY commented:
### Merge activity

* **Sep 18, 5:11 PM UTC**: The merge label 'graphite: merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown) once it meets the requirements.
* **Sep 18, 5:11 PM UTC**: `IWANABETHATGUY` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown).
* **Sep 18, 5:20 PM UTC**: Merged by the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown).

Brooooooklyn committed:
perf(rolldown): take advantage of tokio blocking threads (#6270)

closes https://github.com/rolldown/rolldown/issues/3754

Benchmark in apps/10000:

### `1.0.0-beta.38`

```
❯ hyperfine --warmup 3 --runs 10 --command-name "rolldown-latest" "../../node_modules/.bin/rolldown build --config rolldown.config.mjs"
Benchmark 1: rolldown-latest
  Time (mean ± σ):     553.6 ms ±  21.4 ms    [User: 1107.1 ms, System: 2948.2 ms]
  Range (min … max):   518.7 ms … 579.5 ms    10 runs
```

### `current`

```
❯ hyperfine --warmup 3 --runs 10 --command-name "rolldown-latest" "../../node_modules/.bin/rolldown build --config rolldown.config.mjs"
Benchmark 1: rolldown-latest
  Time (mean ± σ):     458.1 ms ±   6.5 ms    [User: 1052.2 ms, System: 1092.4 ms]
  Range (min … max):   447.5 ms … 468.8 ms    10 runs
```

I don't know why `codspeed` crashes, I added a feature for `codspeed` so that under `codspeed` it doesn't use `spawn_blocking`.
